chapters/packages.tex

```diff
@@ -282,51 +266,42 @@ \section{External Resources}\label{external-resources}
-segment of the path shall not be the name of a class in the package given by the first segment.  As above for \filename{Modelica 3.2.1/package.mo}, i.e., resources starting from
-\filename{Modelica 3.2.1}, and \filename{modelica:/Modelica.Mechanics/C.jpg} is \filename{Modelica 3.2.1/Mechanics/C.jpg} -- regardless of whether \lstinline!Modelica.Mechanics!
-is stored in \filename{Modelica 3.2.1/package.mo}, \filename{Modelica 3.2.1/Mechanics.mo}, or \filename{Modelica 3.2.1/Mechanics/package.mo}.
-
-For a Modelica-package stored as a single file, \filename{A.mo}, the resource
-\filename{modelica:/A/C.jpg} refers to a file \filename{C.jpg} stored in the same
-directory as \filename{A.mo}, but using resources in this variant is not
-recommended since multiple packages will share resources.
-
-In case the name of the class contains quoted identifiers, the single-quote `\lstinline!`!'
-and any reserved characters (`\lstinline!:!', `\lstinline!/!', `\lstinline!?!', `\lstinline!\#!', `\lstinline![!',
-`\lstinline!]!', `\lstinline!@!', `\lstinline!!!', `\lstinline[mathescape=false]!\$!', `\lstinline!\&!', `\lstinline!(!', `\lstinline!)!', `\lstinline!*!', `\lstinline!+!',
-`\lstinline!,!', `\lstinline!;!', `\lstinline!=!') should be percent-encoded as normal in URIs.
+\filename{modelica:/A/Resources/C.jpg} should be stored in the file \filename{A/Resources/C.jpg}, it is not recommend to use \filename{modelica:/A.B/C.jpg} for referencing resources; it could be stored in the file \filename{A/B/C.jpg} -- which is counter-intuitive if \lstinline!A.B! is stored together with \lstinline!A!.
+When Modelica packages are stored in other formats a similar mapping should be defined, such that a resource with a given path should be unique for any package name that precedes it.
+The second segment of the path shall not be the name of a class in the package given by the first segment.
+As above for \filename{Modelica 3.2.1/package.mo}, i.e., resources starting from \filename{Modelica 3.2.1}, and \filename{modelica:/Modelica.Mechanics/C.jpg} is \filename{Modelica 3.2.1/Mechanics/C.jpg} -- regardless of whether \lstinline!Modelica.Mechanics! is stored in \filename{Modelica 3.2.1/package.mo}, \filename{Modelica 3.2.1/Mechanics.mo}, or \filename{Modelica 3.2.1/Mechanics/package.mo}.
+
+When mapping a Modelica URI to a file-system path, the file-system path shall end in a directory separator if and only if the URI path ends in the segment separator `\lstinline!/!'.
+For example, if \filename{modelica:/A/Resources} maps to \filename{A/Resources}, then \filename{modelica:/A/Resources/} maps to \filename{A/Resources/}, and vice versa.
+
+\begin{nonnormative}
+The use of a trailing segment separator is recommended when the resource is a directory and the file-system path will be prepended to relative file paths within the directory.
+\end{nonnormative}
```

Comment:
```suggestion
\begin{nonnormative}
The use of a trailing segment separator is recommended when the resource is a directory and the file-system path will be prepended to relative file paths within the directory.
If possible use URIs for specific files or specific sub-directories instead of appending relative paths to a generic URI as \filename{modelica:/A/Resources/} as the latter can cause the entire directory to be copied.
\end{nonnormative}
```
Or in short in order of preference:

- `\filename{modelica:/A/Resources/Images/a.png}`
- `\filename{modelica:/A/Resources/Images/}`
- `\filename{modelica:/A/Resources/}`
- `\filename{modelica:/A/Resources}`
(and seriously don't use `\filename{modelica:/A/}`)
(This could also be a separate PR.)